File: training-data/objectivec/LeapYear.m
#import <Foundation/Foundation.h>

BOOL isLeapYear(NSInteger year) {
    return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0;
}

int main(int argc, const char *argv[]) {
    @autoreleasepool {
        NSLog(@"%d", isLeapYear(2000));
        NSLog(@"%d", isLeapYear(1900));
        NSLog(@"%d", isLeapYear(2024));
    }
    return 0;
}

File: training-data/prolog/leap_year.pl
is_leap_year(Year) :-
    (0 is Year mod 4, \+ (0 is Year mod 100)) ; 0 is Year mod 400.

:- (is_leap_year(2000) -> writeln(true) ; writeln(false)).
:- (is_leap_year(1900) -> writeln(true) ; writeln(false)).
:- (is_leap_year(2024) -> writeln(true) ; writeln(false)).
